perun/logic/temp.py

```diff
@@ -0,0 +1,653 @@
+
+    :param str file_path: the path to the temporary file
+    :param content: the new content of the temporary file
+    :param bool json_format: if True, the content will be formatted as json
+    :param bool protect: if True, the file will have the protected status
+    :param bool compress: if True, the content will be compressed
+    """
+    file_path = _join_to_tmp_path(file_path)
+    # Make sure that the directory hierarchy for the file exists
+    touch_temp_dir(os.path.dirname(file_path))
+    _write_to_temp(file_path, content, json_format, protect, compress)
+
+
+def get_temp(file_path):
+    """Gets the content of the temporary file 'file_path'. An exception is raised if the file does
+    not exist. None is returned if the file could not have been read, is corrupted, has
+    inconsistent or invalid index properties.
+
+    For details regarding the path format, see the module docstring.
+
+    :param str file_path: the path to the temporary file to read
+
+    :return str or None: the file content or None if an error occurred
+    """
+    # Check the existence of the file and obtain its properties
+    file_path = _join_to_tmp_path(file_path)
+    _is_tmp_file(file_path)
+    json_format, _, compressed = _get_index_entry(file_path)
+    try:
+        with open(file_path, 'rb' if compressed else 'r') as tmp_handle:
+            # Take care of possible compression
+            if compressed:
+                content = store.read_and_deflate_chunk(tmp_handle)
+            else:
+                content = tmp_handle.read()
+            # Parse the json-formatted files
+            if json_format:
+                content = json.loads(content)
+        return content
+    # Handle possible errors
+    except (OSError, ValueError, zlib.error):
+        return None
+
+
+def reset_temp(file_path):
+    """Clears the content and resets properties of the temporary file 'file_path'
+
+    For details regarding the path format, see the module docstring.
+
+    :param str file_path: the path to the temporary file
+    """
+    file_path = _join_to_tmp_path(file_path)
+    _is_tmp_file(file_path)
+    with open(file_path, 'w'):
+        pass
+    _add_to_index(file_path, False, False, False)
+
+
+def list_all_temps(root=None):
+    """Provides a list of all the temporary files in the 'root' folder and all its subfolders.
+
+    For details regarding the path format, see the module docstring.
+
+    If the 'root' is not provided or None, the whole tmp/ folder is considered as the root.
+
+    :param str root: the path to the root folder
+
+    :return list: paths to all the files in the 'root' folder hierarchy
+    """
+    # Get the correct root
+    if root is None:
+        root = pcs.get_tmp_directory()
+    else:
+        root = _join_to_tmp_path(root)
+    _is_tmp_dir(root)
+    # Omit the index file
+    return [os.path.join(dirpath, file) for dirpath, _, files in os.walk(root) for file in files
```

Comment:
Suggestion: reformat?
`return [`
`   os.path...`
`]`

Reply:
Solved, however it still does not fit on one line :)

Reply:
Erm, I meant it like this (should have format it like that from start ;) )
```suggestion
    return [
        os.path.join(dirpath, file) for dirpath, _, files in os.walk(root) for file in files
    ]
```

Reply:
Yes, that's exactly how I modified it, however there is one more condition in the second `for` so the list comprehension is still broken into two lines, but it looks a bit better :+1: